hugr-py/src/hugr/std/int.py

```diff
@@ -44,27 +50,39 @@ def to_value(self) -> val.Extension:
         return val.Extension("int", INT_T, self.v)
 
 
-@dataclass(frozen=True)
-class IntOps(Custom):
-    """Base class for integer operations."""
-
-    extension: tys.ExtensionId = "arithmetic.int"
-
-
-_ARG_I32 = tys.BoundedNatArg(n=5)
+OPS_EXTENSION: tys.ExtensionId = "arithmetic.int"
 
 
 @dataclass(frozen=True)
-class _DivModDef(IntOps):
+class _DivModDef(AsCustomOp):
     """DivMod operation, has two inputs and two outputs."""
 
-    num_out: int = 2
-    extension: tys.ExtensionId = "arithmetic.int"
-    op_name: str = "idivmod_u"
-    signature: tys.FunctionType = field(
-        default_factory=lambda: tys.FunctionType(input=[INT_T] * 2, output=[INT_T] * 2)
-    )
-    args: list[tys.TypeArg] = field(default_factory=lambda: [_ARG_I32, _ARG_I32])
+    op_name: ClassVar[str] = "idivmod_u"
+    arg1: int = 5
+    arg2: int = 5
+
+    def to_custom(self) -> Custom:
+        return Custom(
+            "idivmod_u",
+            tys.FunctionType(
+                input=[int_t(self.arg1)] * 2, output=[int_t(self.arg2)] * 2
+            ),
+            extension=OPS_EXTENSION,
+            args=[tys.BoundedNatArg(n=self.arg1), tys.BoundedNatArg(n=self.arg2)],
+        )
+
+    @classmethod
+    def from_custom(cls, custom: Custom) -> Self | None:
+        if not (custom.extension == OPS_EXTENSION and custom.op_name == cls.op_name):
+            return None
+        match custom.args:
+            case [tys.BoundedNatArg(n=a1), tys.BoundedNatArg(n=a2)]:
+                return cls(arg1=a1, arg2=a2)
+            case _:
+                return None
```

Comment:
Should this raise an error? If the name and extension match then we should always be able to extract the op